numba/tests/test_remove_dead.py

```diff
@@ -3,17 +3,32 @@
 # SPDX-License-Identifier: BSD-2-Clause
 #
 
+import numba
 from numba import compiler, typing
+from numba.compiler import compile_isolated, Flags
 from numba.targets import cpu
 from numba import types
 from numba.targets.registry import cpu_target
 from numba import config
 from numba.annotations import type_annotations
-from numba.ir_utils import copy_propagate, apply_copy_propagate, get_name_var_table, remove_dels, remove_dead
+from numba.ir_utils import (copy_propagate, apply_copy_propagate,
+                            get_name_var_table, remove_dels, remove_dead,
+                            remove_call_handlers)
 from numba import ir
 from numba import unittest_support as unittest
 import numpy as np
 
+# use xxnrm2 to test call a C function with ctypes
+from numba.targets.linalg import _BLAS
```

Comment:
The use BLAS needs guard code around it, at present if SciPy is not available this test module simply fails to load. Perhaps use a `try` with `numba.targets.linalg.ensure_blas` and `numba.tests.test_linalg.needs_blas` as a decorator on the appropriate test methods?

Reply:
Good point. Moved this code to the actual test added `needs_blas` to guard it.